### ISSUE TITLE
Expand All Folders Button

### ISSUE THREAD

**Issue by dangus21** [microsoft/vscode#221596]
Hello VSCode team!
At the present time there is no way to recursively expand all folders, the same way we can collapse them.

Request:
A button next to the collapse that already exists that allows to expand the folders.

PR #221597

**Comment by gjsjohnmurray** [comment]
Related to https://github.com/microsoft/vscode/issues/66811 which now points to #66971

Recursively resolving the File Explorer tree has been debated before. See https://github.com/microsoft/vscode/issues/155570#issuecomment-1189513421 for one of the challenges it presents.

**Comment by dangus21** [comment]
i don't see much about details on why we can't have this feature at this moment, the issue is 2019 and the comment seems to be relating to a bug that is not related to this feature

**Comment by lramos15** [comment]
@dangus21 The comments there still do apply. Can you please give more information about the use case you're trying to solve with expand all? What would be the need of expanding a ton of files in a large repository

**Comment by dangus21** [comment]
recently i was working on a directory with a bunch of folders that all had a subfolder and a config file, and another type of config file was generated into one of them. i missed what the file was since it was being randomized and i had to just about open 50 folders to find where said file was.. this was my use case, but i'm sure there are other people that could possibly benefit from this

**Comment by ericchase** [comment]
the issue mentioned is pretty easy to work around by adding a depth to the recursive expansion. it can even be 1; simply being able to expand folders 1 depth at a time would be a massive utility compared to... nothing. this would be probably be an advanced feature, but i think there are tons of common use cases that shouldn't simply be neglected just because a folder might be too big.

a while back i spent a night setting up the project and then implementing the above (https://github.com/microsoft/vscode/issues/66971#issuecomment-2134261325). personally, i'd find it extremely hard to believe that there is anyone who has never ran into a situation in which this kind of feature would have been useful. but there doesn't seem to be much interest around it

**Comment by ericchase** [comment]
guess we'll try again in 5 more years

**Comment by dangus21** [comment]
> guess we'll try again in 5 more years

we still have 10 days to go 😭

**Comment by robertkirkman** [comment]
Hello, I would like this very much.

I really need this because I'm trying to expand all folders of a project so I can look at all the files in an indented tree structure similar to the `tree` command, but as a GUI. This is important for me to read and understand the filenames.

I hoped that clicking this button twice would expand all folders instead of collapsing all of them like clicking it once does, but unfortunately, that's not what happens - all folders stay collapsed.

<img width="124" height="51" alt="Image" src="https://github.com/user-attachments/assets/7062ed83-4c58-4e46-b737-da6ae474b8b1" />

**Comment by robertkirkman** [comment]
Regarding these messages in one of the linked threads about the reasons why this button isn't planned,

I would like to request, if it's not too difficult to do so, please just expand to 5-folders depth and 5-folder-symbolic-links depth by default, and allow both to be adjusted in settings, as two separate options, one to allow resolving an integer up to unlimited number of folder symbolic links, and the other option to allow expanding an integer up to an unlimited number of nested folders.

<img width="1523" height="900" alt="Image" src="https://github.com/user-attachments/assets/063c6cd7-a7ee-4a38-a218-5689a8eee951" />

I would also like to ask about the part that describes a problem that "when initially opening a workspace, expand all doesn't do anything as we have not resolved any nodes". I don't really understand what this explanation means, but if someone else knows, I would be very grateful if you could explain what that means in simpler terms, and how it is an issue for having an expand-all button in the folder pane.


Sometimes I need to open very large files and very large project folders, and I'm actually very happy with vscode because it performs very well with these kinds of projects compared to other editors. I just think this feature about the expand-all button in the folder pane is something that would be nice to have, to save just a little bit of time.
